tests/Integration/Classes/CartRuleTest.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Classes;
+
+use CartRule;
+use Customer;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShopBundle\Install\DatabaseDump;
+use Tools;
+
+class CartRuleTest extends TestCase
+{
+    /**
+     * @var Customer
+     */
+    public $dummyCustomer;
+
+    /**
+     * @var Configuration
+     */
+    public $configuration;
+
+    /**
+     * @var int
+     */
+    public $defaultLanguageId;
```

Comment:
```suggestion
    /**
     * @var Customer
     */
    protected $dummyCustomer;

    /**
     * @var Configuration
     */
    protected $configuration;

    /**
     * @var int
     */
    protected $defaultLanguageId;
```

Clearly no need to put test properties in public :thinking: